Tests/ApolloCacheDependentTests/WatchQueryTests.swift

```diff
@@ -269,7 +269,7 @@ class WatchQueryTests: XCTestCase, CacheDependentTesting {
     }
     
     runActivity("Fetch same query from server with different argument but returning same object with changed data") { _ in
-      let serverRequestExpectation = server.expect(HeroNameWithIdQuery.self) { request in
+      let serverRequestExpectation = server.expect(HeroNameWithIdQuery.self) { request -> JSONObject in
```

Comment:
I think this stuff was async to help test our async loading - @martijnwalraven care to comment here?

Reply:
Not entirely sure what you mean. This just seems to add a type annotation?

Reply:
This is a type annotation to avoid ambiguity in the type-checker due to the overload. Changing to the throwing form as suggested by Martijn will absolve the need of this change.